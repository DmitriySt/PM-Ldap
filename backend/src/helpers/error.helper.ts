import { ValidationError } from '@nestjs/common';

interface IValidationPipeError {
  name: string;
  errors: string[];
}
export const validationPipeError = (
  errors: ValidationError[],
): IValidationPipeError[] => {
  return errors.map((error) => {
    const errors = [];
    for (const key in error.constraints) {
      if (error.constraints.hasOwnProperty(key)) {
        errors.push(error.constraints[key]);
      }
    }

    return {
      name: error.property,
      errors,
    };
  });
};
