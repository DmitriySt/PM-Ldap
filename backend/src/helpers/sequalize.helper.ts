import sequelize from 'sequelize';

export function dateQuery(col: string, table?: string) {
  col = table ? `"${table}"."${col}"` : `"${col}"`;
  return sequelize.literal(
    `to_char(${col} AT TIME ZONE 'Europe/Moscow', 'dd.mm.YYYY HH24:mi')`,
  );
}
