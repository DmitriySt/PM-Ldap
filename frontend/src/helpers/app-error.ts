import { AxiosError } from "axios";

export interface IValidationErrors {
  name: string;
  errors: string[];
}

export interface IError {
  code: number;
  message: string;
  validationErrors?: IValidationErrors[];
}

export const AppError = (e: unknown): IError => {
  const error: IError = {
    code: 1,
    message: "",
  };
  if (typeof e === "string") {
    error.message = e.toUpperCase();
  } else if (e instanceof AxiosError) {
    error.code = e?.response?.status ? e.response.status : 1;
    error.message = e?.response?.data.message
      ? e.response.data.message
      : e.message;
    if (e?.response?.data.validationErrors) {
      error.validationErrors = e.response.data.validationErrors;
    }
  } else if (e instanceof Error) {
    error.message = e.message;
  }
  return error;
};
