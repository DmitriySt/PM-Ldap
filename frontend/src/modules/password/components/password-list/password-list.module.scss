.layout {
  display: flex;
  flex-direction: column;
  height: 100%;
  border-right: #dce0e5 1px solid;
  background-color: #ffffff;
}

.actionPanel{
  padding: 10px ;

  .buttons{
    display: flex;
    justify-content: center;
    align-content: center;
  }

}

.listLayout{
  overflow: auto;
  position: relative;
  height: 100%;

  scrollbar-width: thin;
  scrollbar-color: #bcbdc2 transparent;
  scrollbar-gutter: stable;

  &::-webkit-scrollbar {
    height: 12px;
    width: 12px;
  }



  &::-webkit-scrollbar-thumb {
    background-color: #bcbdc2;
    border-radius: 5px;
    border: 3px solid #FFFFFF;
  }

}

.list{
  padding: 10px 0 10px 10px;
  position: absolute;
  left: 0;
  right: 0;
}



.listItem{
  padding: 10px;
  border-radius: 5px;

  &:hover{
    background-color: #f5f5f5;
  }

  &.active{
    background-color: #1677ff;
    color: #FFFFFF;
  }
}