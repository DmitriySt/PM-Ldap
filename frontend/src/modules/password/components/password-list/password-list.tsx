import { faMagnifyingGlass, faPlus } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { Button, Col, Input, Row } from "antd";
import classNames from "classnames";
import { useDebounce } from "hooks/useDebounce";
import { usePasswordGroup } from "modules/password/hooks/usePasswordGroup";
import { IPassword } from "modules/password/types/password-types";
import { FC, startTransition, useMemo, useState } from "react";
import styles from "./password-list.module.scss";

export interface PasswordListProps {
  passwordList: IPassword[];
  currentRow: IPassword;
  onChange: (password: IPassword) => void;
  onNewRecord: () => void;
}

const filteredPasswords = (passwordList: IPassword[], searchString: string) => {
  if (searchString.length === 0) {
    return passwordList;
  }
  const _searchString = new RegExp(searchString, "i");
  return passwordList.filter((item) => {
    const searchFields: (keyof IPassword)[] = [
      "name",
      "login",
      "url",
      "description",
    ];
    for (const field of searchFields) {
      if (_searchString.test(item[field].toString())) {
        return true;
      }
    }
    return false;
  });
};

export const PasswordList: FC<PasswordListProps> = ({
  passwordList,
  onChange,
  onNewRecord,
  currentRow,
}) => {
  const { passwordGroup } = usePasswordGroup();
  const [searchString, setSearchString] = useState<string>("");
  const searchStringDebounced = useDebounce(searchString, 500);
  const list = useMemo<IPassword[]>(
    () => filteredPasswords(passwordList, searchStringDebounced),
    [passwordList, searchStringDebounced],
  );

  const handleSearch = (e: React.ChangeEvent<HTMLInputElement>) => {
    const _searchString = e.target.value;
    startTransition(() => setSearchString(_searchString));
  };

  return (
    <div className={styles.layout}>
      <Row className={styles.actionPanel} wrap={false} align={"middle"}>
        <Col flex={"auto"}>
          <Input
            placeholder="Поиск"
            name="search"
            prefix={<FontAwesomeIcon icon={faMagnifyingGlass} />}
            allowClear
            onChange={handleSearch}
          />
        </Col>
        <Col flex="40px" className={styles.buttons}>
          <Button
            size="small"
            type="primary"
            icon={<FontAwesomeIcon icon={faPlus} />}
            title="Добавить запись"
            onClick={onNewRecord}
            disabled={passwordGroup.key === 0}
          ></Button>
        </Col>
      </Row>
      <div className={styles.listLayout}>
        <div className={styles.list}>
          {list.map((item) => (
            <div
              key={item.id}
              className={classNames(
                styles.listItem,
                currentRow.id === item.id ? styles.active : "",
              )}
              onClick={() => onChange(item)}
            >
              <div>{item.name}</div>
              <div>{item.login}</div>
            </div>
          ))}
        </div>
      </div>
    </div>
  );
};
