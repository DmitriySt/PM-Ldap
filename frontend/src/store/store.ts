import {
  combineReducers,
  configureStore,
  PreloadedState,
} from "@reduxjs/toolkit";
import { authReducer } from "modules/auth";
import { passwordGroupReducer } from "modules/password";

const rootReducer = combineReducers({
  authReducer,
  passwordGroupReducer,
});

export const setupStore = (preloadedState?: PreloadedState<RootState>) => {
  return configureStore({
    reducer: rootReducer,
    preloadedState,
    devTools: process.env.NODE_ENV === "development",
  });
};

export type RootState = ReturnType<typeof rootReducer>;
export type AppStore = ReturnType<typeof setupStore>;
export type AppDispatch = AppStore["dispatch"];
export const store = setupStore();
