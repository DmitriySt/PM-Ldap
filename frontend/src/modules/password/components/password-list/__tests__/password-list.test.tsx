import { screen, waitFor } from "@testing-library/react";
import { userEvent } from "@testing-library/user-event";
import { renderWithProviders } from "helpers/test-helper";
import { PasswordList } from "modules/password/components/password-list/password-list";
import {
  groupTreeData,
  passwordEmptyStub,
  passwordListStub,
} from "modules/password/tests/password-group.stub";
import { setupStore } from "store/store";

describe("password-list", () => {
  it("renders empty correctly", async () => {
    const store = setupStore({
      passwordGroupReducer: {
        groupTreeOpen: false,
        groupTree: {
          isLoading: false,
          data: groupTreeData,
        },
        passwordGroup: {
          key: 0,
          title: "",
          parentId: 0,
        },
      },
    });

    const view = renderWithProviders(
      <PasswordList
        currentRow={passwordEmptyStub}
        onChange={() => null}
        onNewRecord={() => null}
        passwordList={[]}
      />,
      { store },
    );

    expect(view).toMatchSnapshot();

    const addButton = screen.getByRole("button", {
      name: /добавить запись/i,
    });

    expect(addButton).toBeInTheDocument();
    expect(addButton).toBeDisabled();
  });

  it("list functional", async () => {
    const store = setupStore({
      passwordGroupReducer: {
        groupTreeOpen: false,
        groupTree: {
          isLoading: false,
          data: groupTreeData,
        },
        passwordGroup: groupTreeData[0],
      },
    });

    const onChange = jest.fn(() => true);
    const onNewRecord = jest.fn();

    const view = renderWithProviders(
      <PasswordList
        currentRow={passwordEmptyStub}
        onChange={onChange}
        onNewRecord={onNewRecord}
        passwordList={passwordListStub}
      />,
      { store },
    );

    const addButton = screen.getByRole("button", {
      name: /добавить запись/i,
    });

    expect(addButton).toBeInTheDocument();
    expect(addButton).toBeEnabled();

    expect(view).toMatchSnapshot();

    await userEvent.click(addButton);
    expect(onNewRecord).toHaveBeenCalledTimes(1);

    expect(
      await screen.findByText(passwordListStub[0].name),
    ).toBeInTheDocument();
    expect(
      await screen.findByText(passwordListStub[0].login),
    ).toBeInTheDocument();

    const searchInput = await screen.findByPlaceholderText(/поиск/i);
    await userEvent.type(searchInput, "name3");
    await waitFor(() => {
      expect(
        screen.queryByText(passwordListStub[0].login),
      ).not.toBeInTheDocument();
    });

    await userEvent.click(screen.getByText(passwordListStub[2].login));
    expect(onChange).toHaveBeenCalledTimes(1);
    expect(onChange).toHaveBeenCalledWith(passwordListStub[2]);
  });
});
