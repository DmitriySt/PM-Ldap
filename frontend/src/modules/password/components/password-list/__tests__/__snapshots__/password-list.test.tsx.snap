// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`password-list list functional 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="layout"
      >
        <div
          class="ant-row ant-row-no-wrap ant-row-middle actionPanel css-dev-only-do-not-override-dkbvqv"
        >
          <div
            class="ant-col css-dev-only-do-not-override-dkbvqv"
            style="flex-basis: auto; min-width: 0;"
          >
            <span
              class="ant-input-affix-wrapper css-dev-only-do-not-override-dkbvqv css-dev-only-do-not-override-dkbvqv"
            >
              <span
                class="ant-input-prefix"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-magnifying-glass "
                  data-icon="magnifying-glass"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"
                    fill="currentColor"
                  />
                </svg>
              </span>
              <input
                class="ant-input css-dev-only-do-not-override-dkbvqv"
                name="search"
                placeholder="Поиск"
                type="text"
                value=""
              />
              <span
                class="ant-input-suffix"
              >
                <span
                  class="ant-input-clear-icon ant-input-clear-icon-hidden"
                  role="button"
                  tabindex="-1"
                >
                  <span
                    aria-label="close-circle"
                    class="anticon anticon-close-circle"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="close-circle"
                      fill="currentColor"
                      fill-rule="evenodd"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M512 64c247.4 0 448 200.6 448 448S759.4 960 512 960 64 759.4 64 512 264.6 64 512 64zm127.98 274.82h-.04l-.08.06L512 466.75 384.14 338.88c-.04-.05-.06-.06-.08-.06a.12.12 0 00-.07 0c-.03 0-.05.01-.09.05l-45.02 45.02a.2.2 0 00-.05.09.12.12 0 000 .07v.02a.27.27 0 00.06.06L466.75 512 338.88 639.86c-.05.04-.06.06-.06.08a.12.12 0 000 .07c0 .03.01.05.05.09l45.02 45.02a.2.2 0 00.09.05.12.12 0 00.07 0c.02 0 .04-.01.08-.05L512 557.25l127.86 127.87c.04.04.06.05.08.05a.12.12 0 00.07 0c.03 0 .05-.01.09-.05l45.02-45.02a.2.2 0 00.05-.09.12.12 0 000-.07v-.02a.27.27 0 00-.05-.06L557.25 512l127.87-127.86c.04-.04.05-.06.05-.08a.12.12 0 000-.07c0-.03-.01-.05-.05-.09l-45.02-45.02a.2.2 0 00-.09-.05.12.12 0 00-.07 0z"
                      />
                    </svg>
                  </span>
                </span>
              </span>
            </span>
          </div>
          <div
            class="ant-col buttons css-dev-only-do-not-override-dkbvqv"
            style="flex: 0 0 40px; min-width: 0;"
          >
            <button
              class="ant-btn css-dev-only-do-not-override-dkbvqv ant-btn-primary ant-btn-sm ant-btn-icon-only"
              title="Добавить запись"
              type="button"
            >
              <span
                class="ant-btn-icon"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-plus "
                  data-icon="plus"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M256 80c0-17.7-14.3-32-32-32s-32 14.3-32 32V224H48c-17.7 0-32 14.3-32 32s14.3 32 32 32H192V432c0 17.7 14.3 32 32 32s32-14.3 32-32V288H400c17.7 0 32-14.3 32-32s-14.3-32-32-32H256V80z"
                    fill="currentColor"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
        <div
          class="listLayout"
        >
          <div
            class="list"
          >
            <div
              class="listItem"
            >
              <div>
                name1
              </div>
              <div>
                login1
              </div>
            </div>
            <div
              class="listItem"
            >
              <div>
                name2
              </div>
              <div>
                login2
              </div>
            </div>
            <div
              class="listItem"
            >
              <div>
                name3
              </div>
              <div>
                login3
              </div>
            </div>
            <div
              class="listItem"
            >
              <div>
                name4
              </div>
              <div>
                login4
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="layout"
    >
      <div
        class="ant-row ant-row-no-wrap ant-row-middle actionPanel css-dev-only-do-not-override-dkbvqv"
      >
        <div
          class="ant-col css-dev-only-do-not-override-dkbvqv"
          style="flex-basis: auto; min-width: 0;"
        >
          <span
            class="ant-input-affix-wrapper css-dev-only-do-not-override-dkbvqv css-dev-only-do-not-override-dkbvqv"
          >
            <span
              class="ant-input-prefix"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-magnifying-glass "
                data-icon="magnifying-glass"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"
                  fill="currentColor"
                />
              </svg>
            </span>
            <input
              class="ant-input css-dev-only-do-not-override-dkbvqv"
              name="search"
              placeholder="Поиск"
              type="text"
              value=""
            />
            <span
              class="ant-input-suffix"
            >
              <span
                class="ant-input-clear-icon ant-input-clear-icon-hidden"
                role="button"
                tabindex="-1"
              >
                <span
                  aria-label="close-circle"
                  class="anticon anticon-close-circle"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="close-circle"
                    fill="currentColor"
                    fill-rule="evenodd"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M512 64c247.4 0 448 200.6 448 448S759.4 960 512 960 64 759.4 64 512 264.6 64 512 64zm127.98 274.82h-.04l-.08.06L512 466.75 384.14 338.88c-.04-.05-.06-.06-.08-.06a.12.12 0 00-.07 0c-.03 0-.05.01-.09.05l-45.02 45.02a.2.2 0 00-.05.09.12.12 0 000 .07v.02a.27.27 0 00.06.06L466.75 512 338.88 639.86c-.05.04-.06.06-.06.08a.12.12 0 000 .07c0 .03.01.05.05.09l45.02 45.02a.2.2 0 00.09.05.12.12 0 00.07 0c.02 0 .04-.01.08-.05L512 557.25l127.86 127.87c.04.04.06.05.08.05a.12.12 0 00.07 0c.03 0 .05-.01.09-.05l45.02-45.02a.2.2 0 00.05-.09.12.12 0 000-.07v-.02a.27.27 0 00-.05-.06L557.25 512l127.87-127.86c.04-.04.05-.06.05-.08a.12.12 0 000-.07c0-.03-.01-.05-.05-.09l-45.02-45.02a.2.2 0 00-.09-.05.12.12 0 00-.07 0z"
                    />
                  </svg>
                </span>
              </span>
            </span>
          </span>
        </div>
        <div
          class="ant-col buttons css-dev-only-do-not-override-dkbvqv"
          style="flex: 0 0 40px; min-width: 0;"
        >
          <button
            class="ant-btn css-dev-only-do-not-override-dkbvqv ant-btn-primary ant-btn-sm ant-btn-icon-only"
            title="Добавить запись"
            type="button"
          >
            <span
              class="ant-btn-icon"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-plus "
                data-icon="plus"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 80c0-17.7-14.3-32-32-32s-32 14.3-32 32V224H48c-17.7 0-32 14.3-32 32s14.3 32 32 32H192V432c0 17.7 14.3 32 32 32s32-14.3 32-32V288H400c17.7 0 32-14.3 32-32s-14.3-32-32-32H256V80z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
      <div
        class="listLayout"
      >
        <div
          class="list"
        >
          <div
            class="listItem"
          >
            <div>
              name1
            </div>
            <div>
              login1
            </div>
          </div>
          <div
            class="listItem"
          >
            <div>
              name2
            </div>
            <div>
              login2
            </div>
          </div>
          <div
            class="listItem"
          >
            <div>
              name3
            </div>
            <div>
              login3
            </div>
          </div>
          <div
            class="listItem"
          >
            <div>
              name4
            </div>
            <div>
              login4
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`password-list renders empty correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="layout"
      >
        <div
          class="ant-row ant-row-no-wrap ant-row-middle actionPanel css-dev-only-do-not-override-dkbvqv"
        >
          <div
            class="ant-col css-dev-only-do-not-override-dkbvqv"
            style="flex-basis: auto; min-width: 0;"
          >
            <span
              class="ant-input-affix-wrapper css-dev-only-do-not-override-dkbvqv css-dev-only-do-not-override-dkbvqv"
            >
              <span
                class="ant-input-prefix"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-magnifying-glass "
                  data-icon="magnifying-glass"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"
                    fill="currentColor"
                  />
                </svg>
              </span>
              <input
                class="ant-input css-dev-only-do-not-override-dkbvqv"
                name="search"
                placeholder="Поиск"
                type="text"
                value=""
              />
              <span
                class="ant-input-suffix"
              >
                <span
                  class="ant-input-clear-icon ant-input-clear-icon-hidden"
                  role="button"
                  tabindex="-1"
                >
                  <span
                    aria-label="close-circle"
                    class="anticon anticon-close-circle"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="close-circle"
                      fill="currentColor"
                      fill-rule="evenodd"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M512 64c247.4 0 448 200.6 448 448S759.4 960 512 960 64 759.4 64 512 264.6 64 512 64zm127.98 274.82h-.04l-.08.06L512 466.75 384.14 338.88c-.04-.05-.06-.06-.08-.06a.12.12 0 00-.07 0c-.03 0-.05.01-.09.05l-45.02 45.02a.2.2 0 00-.05.09.12.12 0 000 .07v.02a.27.27 0 00.06.06L466.75 512 338.88 639.86c-.05.04-.06.06-.06.08a.12.12 0 000 .07c0 .03.01.05.05.09l45.02 45.02a.2.2 0 00.09.05.12.12 0 00.07 0c.02 0 .04-.01.08-.05L512 557.25l127.86 127.87c.04.04.06.05.08.05a.12.12 0 00.07 0c.03 0 .05-.01.09-.05l45.02-45.02a.2.2 0 00.05-.09.12.12 0 000-.07v-.02a.27.27 0 00-.05-.06L557.25 512l127.87-127.86c.04-.04.05-.06.05-.08a.12.12 0 000-.07c0-.03-.01-.05-.05-.09l-45.02-45.02a.2.2 0 00-.09-.05.12.12 0 00-.07 0z"
                      />
                    </svg>
                  </span>
                </span>
              </span>
            </span>
          </div>
          <div
            class="ant-col buttons css-dev-only-do-not-override-dkbvqv"
            style="flex: 0 0 40px; min-width: 0;"
          >
            <button
              class="ant-btn css-dev-only-do-not-override-dkbvqv ant-btn-primary ant-btn-sm ant-btn-icon-only"
              disabled=""
              title="Добавить запись"
              type="button"
            >
              <span
                class="ant-btn-icon"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-plus "
                  data-icon="plus"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M256 80c0-17.7-14.3-32-32-32s-32 14.3-32 32V224H48c-17.7 0-32 14.3-32 32s14.3 32 32 32H192V432c0 17.7 14.3 32 32 32s32-14.3 32-32V288H400c17.7 0 32-14.3 32-32s-14.3-32-32-32H256V80z"
                    fill="currentColor"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
        <div
          class="listLayout"
        >
          <div
            class="list"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="layout"
    >
      <div
        class="ant-row ant-row-no-wrap ant-row-middle actionPanel css-dev-only-do-not-override-dkbvqv"
      >
        <div
          class="ant-col css-dev-only-do-not-override-dkbvqv"
          style="flex-basis: auto; min-width: 0;"
        >
          <span
            class="ant-input-affix-wrapper css-dev-only-do-not-override-dkbvqv css-dev-only-do-not-override-dkbvqv"
          >
            <span
              class="ant-input-prefix"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-magnifying-glass "
                data-icon="magnifying-glass"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"
                  fill="currentColor"
                />
              </svg>
            </span>
            <input
              class="ant-input css-dev-only-do-not-override-dkbvqv"
              name="search"
              placeholder="Поиск"
              type="text"
              value=""
            />
            <span
              class="ant-input-suffix"
            >
              <span
                class="ant-input-clear-icon ant-input-clear-icon-hidden"
                role="button"
                tabindex="-1"
              >
                <span
                  aria-label="close-circle"
                  class="anticon anticon-close-circle"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="close-circle"
                    fill="currentColor"
                    fill-rule="evenodd"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M512 64c247.4 0 448 200.6 448 448S759.4 960 512 960 64 759.4 64 512 264.6 64 512 64zm127.98 274.82h-.04l-.08.06L512 466.75 384.14 338.88c-.04-.05-.06-.06-.08-.06a.12.12 0 00-.07 0c-.03 0-.05.01-.09.05l-45.02 45.02a.2.2 0 00-.05.09.12.12 0 000 .07v.02a.27.27 0 00.06.06L466.75 512 338.88 639.86c-.05.04-.06.06-.06.08a.12.12 0 000 .07c0 .03.01.05.05.09l45.02 45.02a.2.2 0 00.09.05.12.12 0 00.07 0c.02 0 .04-.01.08-.05L512 557.25l127.86 127.87c.04.04.06.05.08.05a.12.12 0 00.07 0c.03 0 .05-.01.09-.05l45.02-45.02a.2.2 0 00.05-.09.12.12 0 000-.07v-.02a.27.27 0 00-.05-.06L557.25 512l127.87-127.86c.04-.04.05-.06.05-.08a.12.12 0 000-.07c0-.03-.01-.05-.05-.09l-45.02-45.02a.2.2 0 00-.09-.05.12.12 0 00-.07 0z"
                    />
                  </svg>
                </span>
              </span>
            </span>
          </span>
        </div>
        <div
          class="ant-col buttons css-dev-only-do-not-override-dkbvqv"
          style="flex: 0 0 40px; min-width: 0;"
        >
          <button
            class="ant-btn css-dev-only-do-not-override-dkbvqv ant-btn-primary ant-btn-sm ant-btn-icon-only"
            disabled=""
            title="Добавить запись"
            type="button"
          >
            <span
              class="ant-btn-icon"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-plus "
                data-icon="plus"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 80c0-17.7-14.3-32-32-32s-32 14.3-32 32V224H48c-17.7 0-32 14.3-32 32s14.3 32 32 32H192V432c0 17.7 14.3 32 32 32s32-14.3 32-32V288H400c17.7 0 32-14.3 32-32s-14.3-32-32-32H256V80z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
      <div
        class="listLayout"
      >
        <div
          class="list"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
